operator/v1/types_console.go

```diff
@@ -106,6 +106,47 @@ type ConsoleCustomization struct {
 	// SVG format preferred
 	// +optional
 	CustomLogoFile configv1.ConfigMapFileReference `json:"customLogoFile,omitempty"`
+	// developerCatalog allows to configure the shown developer catalog categories.
+	// +kubebuilder:validation:Optional
+	// +optional
+	DeveloperCatalog DeveloperConsoleCatalogCustomization `json:"developerCatalog,omitempty"`
+}
+
+// developerCatalog allow the cluster admin to configure developer catalog.
+type DeveloperConsoleCatalogCustomization struct {
+	// categories which are shown the in developer catalog.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Categories []DeveloperConsoleCatalogCategory `json:"categories,omitempty"`
+}
+
+// category for the developer console catalog.
+type DeveloperConsoleCatalogCategoryMeta struct {
+	// ID is an identifier used in the URL to enable deep linking in console.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
```

Comment:
Should we also add max length validation ?
```
// +kubebuilder:validation:MinLength=32
```

or a regexp that would only allow certain chars?
```
// +kubebuilder:validation:Pattern=`...`
```

Reply:
The same for `Label`

Reply:
Sounds good, esp. for ID because we use this in URLs. Defined them now as:

```go
	// ID is an identifier used in the URL to enable deep linking in console.
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MinLength=1
	// +kubebuilder:validation:MinLength=32
	// +kubebuilder:validation:Pattern=`^[A-Za-z0-9-_]+$`
	// +required
	ID string `json:"id"`
	// label defines a category display label.
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MinLength=1
	// +kubebuilder:validation:MinLength=64
	// +required
	Label string `json:"label"`
```